### PR TITLE
add support for 32bit colors in UI

### PR DESCRIPTION
**Pull request by TychoVrahe**
This PR adds support for 32 bit colors in UI. This will enable doing cool stuff like gradients (on displays that support it, currently only available on DISC2).

Some minor bugs are fixed in the underlying C code of the drawing lib.

### CODE REVIEW

File: core/embed/lib/gfx_color.h [hunk 1]
```diff
@@ -22,8 +22,11 @@
 
 #include <stdint.h>
 
+#ifdef UI_COLORS32BIT
```
Comment (cepetr) [inline core/embed/lib/gfx_color.h:25]: I would prefer to maintain consistency with the existing name `GFX_COLOR_32BIT` and suggest using either `UI_COLOR32BIT` (or `UI_COLOR_32BIT`) and the Rust feature `ui_color32bit` (or `ui_color_32bit`).

File: core/embed/rust/src/ui/display/color.rs [hunk 2]
```diff
@@ -1,12 +1,31 @@
...
 impl Color {
+    #[cfg(not(feature = "ui_colors32bit"))]
     pub const fn from_u16(val: u16) -> Self {
         Self(val)
     }
+
+    #[cfg(feature = "ui_colors32bit")]
+    pub const fn from_u16(val: u16) -> Self {
+        Self {
+            r: ((val >> 8) & 0xF8) as u8,
```
Comment (cepetr) [inline core/embed/rust/src/ui/display/color.rs:23]: The code works, but it lacks precision. The value 0xFFFF is converted to 0xF8FCF800. The lower bits should replicate the highest bits value. Please look at `gfx_color16_to_r/g/b()` functions in `gfx_color.h`.

**Review by cepetr — CHANGES_REQUESTED** [review]

File: core/embed/lib/gfx_color.h (hunk 1 above)
Comment (TychoVrahe) [reply core/embed/lib/gfx_color.h]: renamed: https://github.com/trezor/trezor-firmware/commit/92df5bf66d3e6964ec166f0b3e8be1a7643f6c33

File: core/embed/rust/src/ui/display/color.rs (hunk 2 above)
Comment (TychoVrahe) [reply core/embed/rust/src/ui/display/color.rs]: fixed https://github.com/trezor/trezor-firmware/commit/92df5bf66d3e6964ec166f0b3e8be1a7643f6c33

**Pull request merged**